header/p2p/options.go

```diff
@@ -82,6 +90,19 @@ func WithMaxRequestSize[T parameters](size uint64) Option[T] {
 	}
 }
 
+// WithRequestDuration is a functional option that configures the
+// `RequestDuration` parameter.
+func WithRequestDuration[T parameters](duration time.Duration) Option[T] {
+	return func(p *T) {
+		switch t := any(p).(type) {
+		case *ClientParameters:
+			t.RequestDuration = duration
+		case *ServerParameters:
+			t.RequestDuration = duration
```

Comment:
Does the server need it?

Reply:
to limit requests in store

Reply:
https://github.com/celestiaorg/celestia-node/pull/1588

Reply:
Isn't this PR solves it?